### DISCUSSION TITLE
How to switch import library by platform

### DISCUSSION THREAD

**Discussion by natsucurry** [odin-lang/Odin#3438] [Q&A] [answered]
Hello,Thank you for awesome odin. 
I develop desktop and webgl game with Raylib-odin.
When switch import library  by platform, Redeclaration of proc is occurred somehow.
This is my code. 
```odin
when ODIN_BUILD_MODE == .Object {
	import rl   "../lib/raylib"     // web build
} else  {
	import rl   "vendor:raylib"     // desktop build
}
```
ODIN_BUILD_MODE flag is working correctly,but checker is called twice and because of two raylib have same name functions,  Redeclaration of proc is occurred.
When comment out Web or Desktop import,raylib-odin itself is working in either case.
I want to know How to avoid this problem.
Thank you for all help.

**Comment by Kelimion** [comment]
What you could do is use the `*_<platform>.odin` filename convention, i.e.

`rl_wasm.odin`
```odin
import rl "../lib/raylib" // web build
```

`rl_windows.odin`
```odin
import rl "vendor:raylib" // desktop build
```

**Reply by natsucurry** [reply]
This is my `*_<platform>.odin` file suffixes.

`rl_wasm.odin`
```odin
package odingame
import rl "../lib/raylib" 
```
`rl_windows.odin`
```odin
package odingame
import rl "vendor:raylib" 
```
Can I access `rl` of `rl_wasm.odin` or `rl_windows.odin`  from my `main.odin`?

**Reply by natsucurry** [reply]
I tried,but file suffixes approach also redeclaration of proc is occurred.
In another way, I  try build tags.
add `//+build wasm32 `  to `"../lib/raylib.odin"` file just above `package raylib`
add `//+build windows `  to `"vendor:raylib.odin"` file just above `package raylib`

This is my main.odin
```odin
when ODIN_BUILD_MODE == .Object {
	import rl   "../lib/raylib"     // web build
} else  {
	import rl   "vendor:raylib"     // desktop build
}
```
desktop build commands 
```
odin run  src -file
```
web build commands 
```
odin build src -file -out=odin -build-mode:obj -disable-assert -no-bounds-check -show-system-calls -debug --target:freestanding_wasm32
```
Desktop build is succeeded.
Web build is not succeeded.
Because of not ignoring vendor:raylib build, duplicate declaration of 'package raylib' is occurred.
So change build tag `vendor:raylib`  to `//+build ignore`,but the result is the same
Is anything wrong?


By the way,this main.odin removed `when statement` is succeeded on web build.
```odin
import rl   "../lib/raylib"     // web build
//import rl   "vendor:raylib"     // desktop build
```

**Reply by natsucurry** [reply, marked as answer]
I did it.
To add //+build windows  and //+build wasm32 tags each files,the above file suffixes approach is working correctly.

This is my final approach.

game project.
```
odingame/
　├ dist/
　├ lib/
         ├ raylib
                 ├ wasm     //  library directory
                 ├ windows    //  library directory
                 ├ raylib_wasm.odin     // copied from https://github.com/Caedo/raylib_wasm_odin
                 ├ raylib_windows.odin     // copied from vendor:raylib
                 ├ easings_windows.odin     // copied from vendor:raylib
                 └ rlgl_windows.odin     // copied from vendor:raylib
　├ src/
　│　└ main.odin
```
main.odin
```odin
import rl   "../lib/raylib"      // when statement is not needed
```
desktop build commands 
```
odin run  src -file
```
web build commands 
```
odin build src -file -out=odin -build-mode:obj -disable-assert -no-bounds-check -show-system-calls -debug --target:freestanding_wasm32
```
Thank you for your help.
